api/tests/unit/domain/usecases/remove-authentication-method_test.js

```diff
@@ -181,76 +216,45 @@ describe('Unit | UseCase | remove-authentication-method', function () {
       authenticationMethodRepository.findByUserId.resolves(authenticationMethods);
 
       // when
-      await removeAuthenticationMethod({ userId: user.id, type, userRepository, authenticationMethodRepository });
-
-      // then
-      expect(authenticationMethodRepository.removeByUserIdAndIdentityProvider).to.have.been.calledWithExactly({
+      await removeAuthenticationMethod({
         userId: user.id,
-        identityProvider: OidcIdentityProviders.POLE_EMPLOI.code,
+        type,
+        userRepository,
+        authenticationMethodRepository,
       });
-    });
-  });
-
-  context('When type is CNAV', function () {
-    it('should remove CNAV authentication method', async function () {
-      // given
-      const type = OidcIdentityProviders.CNAV.code;
-      const user = domainBuilder.buildUser();
-      userRepository.get.resolves(user);
-      const authenticationMethods = buildAllAuthenticationMethodsForUser(user.id);
-      authenticationMethodRepository.findByUserId.resolves(authenticationMethods);
-
-      // when
-      await removeAuthenticationMethod({ userId: user.id, type, userRepository, authenticationMethodRepository });
 
       // then
       expect(authenticationMethodRepository.removeByUserIdAndIdentityProvider).to.have.been.calledWithExactly({
         userId: user.id,
-        identityProvider: OidcIdentityProviders.CNAV.code,
+        identityProvider: OidcIdentityProviders.POLE_EMPLOI.code,
       });
     });
   });
-
-  context('When type is FWB', function () {
-    it('removes FWB authentication method', async function () {
+  context('When type is any generic oidc', function () {
+    it('should remove type authentication method', async function () {
```

Comment:
La formulation n'est pas encore bonne (je trouve)

```suggestion
    it('removes type authentication method', async function () {
```